aws_lambda_builders/workflows/nodejs_npm/workflow.py

```diff
@@ -26,64 +39,110 @@ class NodejsNpmWorkflow(BaseWorkflow):
         )
-
         npm_copy_npmrc = NodejsNpmrcCopyAction(tar_package_dir, source_dir, osutils=osutils)
 
-        self.actions = [
+        actions = [
             npm_pack,
             npm_copy_npmrc,
             CopySourceAction(tar_package_dir, artifacts_dir, excludes=self.EXCLUDED_FILES),
         ]
 
         if self.download_dependencies:
             # installed the dependencies into artifact folder
-            self.actions.append(NodejsNpmInstallAction(artifacts_dir, subprocess_npm=subprocess_npm))
+            actions.append(NodejsNpmInstallAction(artifacts_dir, subprocess_npm=subprocess_npm))
 
             # if dependencies folder exists, copy or move dependencies from artifact folder to dependencies folder
             # depends on the combine_dependencies flag
             if self.dependencies_dir:
                 # clean up the dependencies folder first
-                self.actions.append(CleanUpAction(self.dependencies_dir))
+                actions.append(CleanUpAction(self.dependencies_dir))
                 # if combine_dependencies is set, we should keep dependencies and source code in the artifact folder
                 # while copying the dependencies. Otherwise we should separate the dependencies and source code
                 if self.combine_dependencies:
-                    self.actions.append(CopyDependenciesAction(source_dir, artifacts_dir, self.dependencies_dir))
+                    actions.append(CopyDependenciesAction(source_dir, artifacts_dir, self.dependencies_dir))
                 else:
-                    self.actions.append(MoveDependenciesAction(source_dir, artifacts_dir, self.dependencies_dir))
+                    actions.append(MoveDependenciesAction(source_dir, artifacts_dir, self.dependencies_dir))
         else:
             # if dependencies folder exists and not download dependencies, simply copy the dependencies from the
             # dependencies folder to artifact folder
             if self.dependencies_dir and self.combine_dependencies:
-                self.actions.append(CopySourceAction(self.dependencies_dir, artifacts_dir))
+                actions.append(CopySourceAction(self.dependencies_dir, artifacts_dir))
             else:
                 LOG.info(
                     "download_dependencies is False and dependencies_dir is None. Copying the source files into the "
                     "artifacts directory. "
                 )
 
-        self.actions.append(NodejsNpmrcCleanUpAction(artifacts_dir, osutils=osutils))
+        actions.append(NodejsNpmrcCleanUpAction(artifacts_dir, osutils=osutils))
+        actions.append(NodejsNpmLockFileCleanUpAction(artifacts_dir, osutils=osutils))
+
+        if self.dependencies_dir:
+            actions.append(NodejsNpmLockFileCleanUpAction(self.dependencies_dir, osutils=osutils))
+
+        return actions
+
+    def actions_with_bundler(self, source_dir, artifacts_dir, bundler_config, osutils, subprocess_npm):
```

Comment:
Are we able to support `self.download_dependencies`, `self.dependencies_dir`, and `self.combine_dependencies` logic in `actions_with_bundler` as well?
https://github.com/aws/aws-lambda-builders/pull/272#discussion_r700834619

Reply:
I followed the logic in this table:

| download_dependencies | dependencies_dir | combine_dependencies | Outcome                                                        | Actions                                                                                  |
|-----------------------|------------------|----------------------|----------------------------------------------------------------|------------------------------------------------------------------------------------------|
| True                  | Specified        | False                | Skip bundling, treat like unbundled build process              | Install/build dependencies  into artifact dir and  copy dependencies only  into dep dir. |
| True                  | Specified        | True                 | Standard esbuild process  and copying bundled code to dep dir. | Install, cleanup dep dir,  esbuild to artifacts dir,  copy artifacts dir to dep dir.     |
| True                  | None             | N/A                  | Standard esbuild process                                       | Install and esbuild                                                                      |
| False                 | N/A              | N/A                  | Only bundle into  artifacts dir                                | esbuild                                                                                  |

I'm not certain about the True, Specified, True scenario, however. Please review and let me know if it needs any changes.